src/SubMenu.js

```diff
@@ -1,42 +1,58 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 
+import AbstractMenu from './AbstractMenu';
 import { cssClasses, hasOwnProp } from './helpers';
 
-export default class SubMenu extends Component {
+export default class SubMenu extends AbstractMenu {
     static propTypes = {
         children: PropTypes.node.isRequired,
         title: PropTypes.node.isRequired,
         className: PropTypes.string,
         disabled: PropTypes.bool,
         hoverDelay: PropTypes.number,
-        rtl: PropTypes.bool
+        rtl: PropTypes.bool,
+        selected: PropTypes.bool,
+        onMouseMove: PropTypes.func.isRequired,
+        onMouseOut: PropTypes.func.isRequired,
+        forceOpen: PropTypes.bool.isRequired,
+        forceClose: PropTypes.func.isRequired,
+        parentKeyNavigationHandler: PropTypes.func.isRequired
```

Comment:
Why is all this stuff marked as required? After upgrading to latest release it's asking me for a lot of props yet seems to work fine without them.

Also would be good to have more documentation about what these are, because frankly I don't know what forceOpen or parentKeyNavigationHandler are supposed to do at first glance :)